crypto/bls/blst/signature.go

```diff
@@ -46,6 +46,36 @@ func SignatureFromBytes(sig []byte) (common.Signature, error) {
 	return &Signature{s: signature}, nil
 }
 
+// MultipleSignaturesFromBytes creates a group of BLS signatures from a LittleEndian 2d-byte slice.
+func MultipleSignaturesFromBytes(multiSigs [][]byte) ([]common.Signature, error) {
+	if features.Get().SkipBLSVerify {
+		return []common.Signature{}, nil
+	}
+	if len(multiSigs) == 0 {
+		return nil, fmt.Errorf("0 signatures provided to the method")
+	}
+	for _, s := range multiSigs {
+		if len(s) != fieldparams.BLSSignatureLength {
+			return nil, fmt.Errorf("signature must be %d bytes", fieldparams.BLSSignatureLength)
+		}
+	}
+	multiSignatures := new(blstSignature).BatchUncompress(multiSigs)
+	if multiSignatures == nil || len(multiSignatures) == 0 {
+		return nil, errors.New("could not unmarshal bytes into signature")
+	}
+	wrappedSigs := make([]common.Signature, 0, len(multiSignatures))
+	for _, signature := range multiSignatures {
+		// Group check signature. Do not check for infinity since an aggregated signature
+		// could be infinite.
+		if !signature.SigValidate(false) {
+			return nil, errors.New("signature not in group")
+		}
+		copiedSig := signature
```

Comment:
why is the copy necessary?

Reply:
The reference to`signature` changes in the next iteration of the loop so you don't want the slice to be referring to changed references internally. Instead of 
```
[]{a,b,c}
```
it becomes
```
[]{c,c,c}
```